src/Tempest/Framework/Testing/Http/TestResponseHelper.php

```diff
@@ -219,6 +219,78 @@ public function assertNotSee(string $search): self
         return $this;
     }
 
+    public function assertViewData(string $key, ?Closure $test = null): self
+    {
+        $data = $this->body->data;
+        $value = $data[$key];
+
+        Assert::assertArrayHasKey(
+            key: $key,
+            array: $data,
+            message: sprintf(
+                'No view data was set for [%s], available view data keys: %s',
+                $key,
+                implode(', ', array_keys($data)),
+            ),
+        );
+
+        if ($test !== null) {
+            $test($data, $value);
```

Comment:
We probably want to fail assertion if the callback returns `false` specifically (not `null`, so failing is opt-in)